src/test/scala/wdl4s/types/WdlPairTypeSpec.scala

```diff
@@ -96,7 +115,34 @@ class WdlPairTypeSpec extends FlatSpec with Matchers {
         |  }
         |}
       """.stripMargin
-    
+
     noException should be thrownBy WdlNamespaceWithWorkflow.load(wdl, Seq.empty)
   }
+
+  it should "coerce a Map into a WdlArray of WdlPair type" in {
+    WdlArrayType(WdlPairType(WdlStringType, WdlIntegerType)).coerceRawValue(stringIntMap) match {
+      case Success(array) => array shouldEqual arrayOfPairs
+      case Failure(f) => fail(s"exception while coercing JsObject: $f")
+    }
+  }
+
+  it should "coerce a JsArray into a WdlArray of WdlPairs" in {
+    val jsArray = JsArray(JsObject("a" -> JsNumber(1)), JsObject("b" -> JsNumber(2)), JsObject("c" -> JsNumber(3)))
+
+    WdlArrayType(WdlPairType(WdlStringType, WdlIntegerType)).coerceRawValue(jsArray) match {
+      case Success(array) => array shouldEqual arrayOfPairs
+      case Failure(f) => fail(s"exception while coercing JsObject: $f")
+    }
+  }
+
+  it should "coerce a complex JsArray into a WdlArray of WdlPairs of WdlArrays" in {
+    val complexJsArray =  JsArray(JsObject("a" -> JsArray(JsNumber(1), JsNumber(11))),
+                                  JsObject("b" -> JsArray(JsNumber(2), JsNumber(21))),
+                                  JsObject("c" -> JsArray(JsNumber(3), JsNumber(31))))
+
```

Comment:
I think we wanted this to be an object with `left` and `right` fields, so that we can have also complex types on the LHS too, e.g. 
```
pairs: [
  {
    left: "a",
    right: [1, 11]
  },
  {
    left: "b",
    right: [2, 22]
  },
  {
    left: "c",
    right: [3, 33]
  },
]
```